packages/typespec-client-generator-core/src/internal-utils.ts

```diff
@@ -302,3 +303,20 @@ export function isNullable(type: Type | SdkServiceOperation): boolean {
 export function createGeneratedName(type: Namespace | Operation, suffix: string): string {
   return `${getCrossLanguageDefinitionId(type).split(".").at(-1)}${suffix}`;
 }
+
+function isOperationBodyType(context: TCGCContext, type: Type, operation?: Operation): boolean {
+  if (type.kind !== "Model") return false;
+  if (!isHttpOperation(context, operation)) return false;
+  const httpBody = operation
+    ? getHttpOperationWithCache(context, operation).parameters.body
+    : undefined;
+  return (
+    !!httpBody &&
```

Comment:
nit: Boolean(httpBody) can be used here, it has the same effect but reads a bit nicer